### ISSUE TITLE
NRE on head request with invalid url

### ISSUE THREAD

**Issue by actislav** [Cysharp/UniTask#108]
```
NullReferenceException: Object reference not set to an instance of an object
Cysharp.Threading.Tasks.UnityWebRequestException..ctor (UnityEngine.Networking.UnityWebRequest unityWebRequest) (at Library/PackageCache/com.cysharp.unitask@c1f75d9ebd/Runtime/UnityWebRequestException.cs:34)
Cysharp.Threading.Tasks.UnityAsyncExtensions+UnityWebRequestAsyncOperationConfiguredSource.MoveNext () (at Library/PackageCache/com.cysharp.unitask@c1f75d9ebd/Runtime/UnityAsyncExtensions.cs:1475)
Cysharp.Threading.Tasks.Internal.PlayerLoopRunner.RunCore () (at Library/PackageCache/com.cysharp.unitask@c1f75d9ebd/Runtime/Internal/PlayerLoopRunner.cs:154)
UnityEngine.Debug:LogException(Exception)
Cysharp.Threading.Tasks.Internal.<>c:<.ctor>b__9_0(Exception) (at Library/PackageCache/com.cysharp.unitask@c1f75d9ebd/Runtime/Internal/PlayerLoopRunner.cs:25)
Cysharp.Threading.Tasks.Internal.PlayerLoopRunner:RunCore() (at Library/PackageCache/com.cysharp.unitask@c1f75d9ebd/Runtime/Internal/PlayerLoopRunner.cs:168)
Cysharp.Threading.Tasks.Internal.PlayerLoopRunner:Update() (at Library/PackageCache/com.cysharp.unitask@c1f75d9ebd/Runtime/Internal/PlayerLoopRunner.cs:127)
Cysharp.Threading.Tasks.Internal.PlayerLoopRunner:Run() (at Library/PackageCache/com.cysharp.unitask@c1f75d9ebd/Runtime/Internal/PlayerLoopRunner.cs:94)
```

```csharp
 var webRequestAsyncOperation = UnityWebRequest.Head(url).SendWebRequest();
 await webRequestAsyncOperation.ToUniTask(progress, cancellationToken: cancellationToken);
```

**Comment by neuecc** [comment]
thanks!
I've confirmed and fixed.

**Comment by neuecc** [comment]
fixed at 2.0.25.
